src/iuse.cpp

```diff
@@ -4242,7 +4243,8 @@ int iuse::shocktonfa_on( player *p, item *it, bool t, const tripoint &pos )
 int iuse::mp3( player *p, item *it, bool, const tripoint & )
 {
     // TODO: avoid item id hardcoding to make this function usable for pure json-defined devices.
-    if( !it->units_sufficient( *p ) ) {
+    if( !it->units_sufficient( *p ) || !( it->has_flag( flag_USE_UPS ) &&
```

Comment:
I think it should be `&&` here. Might be the cause of #47478

Reply:
Just in case - Ping: @Jamuro-g

Reply:
thank you i ll check